main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Projects.FileNesting/NestingRule.cs

```diff
@@ -65,7 +66,7 @@ ProjectFile CheckParentForFile (ProjectFile inputFile, FilePath parentFile, File
 			if (inputFile.FilePath != parentFile && !parentFile.IsDirectory && inDirectory == parentFile.ParentDirectory) {
 				var parent = inputFile.Project.Files.GetFile (parentFile);
 				if (parent != null) {
-					LoggingService.LogInfo ($"Applied rule for nesting {inputFile} under {parentFile}");
+					Debug.WriteLine ($"Applied rule for nesting {inputFile} under {parentFile}");
```

Comment:
To avoid allocating the string when not under a debugger.

```suggestion
					Debug.Print ("Applied rule for nesting {0} under {1}", inputFile, parentFile);
```

Reply:
Nevermind, this is conditionally compiled code.